src/azure-cli/azure/cli/command_modules/iot/commands.py

```diff
@@ -59,13 +59,17 @@ def load_command_table(self, _):  # pylint: disable=too-many-statements
         g.generic_update_command('update', getter_name='iot_dps_get', setter_name='iot_dps_update',
                                  command_type=update_custom_util)
 
-    # iot dps access-policy commands
-    with self.command_group('iot dps access-policy', client_factory=iot_service_provisioning_factory) as g:
-        g.custom_command('list', 'iot_dps_access_policy_list')
-        g.custom_show_command('show', 'iot_dps_access_policy_get')
-        g.custom_command('create', 'iot_dps_access_policy_create', supports_no_wait=True)
-        g.custom_command('update', 'iot_dps_access_policy_update', supports_no_wait=True)
-        g.custom_command('delete', 'iot_dps_access_policy_delete', supports_no_wait=True)
+    # iot dps access-policy commands (Deprecated)
+    with self.command_group('iot dps access-policy',
+                            client_factory=iot_service_provisioning_factory,
+                            deprecate_info=self.deprecate(redirect='iot dps policy',
+                                                          expiration='2.35.0')
```

Comment:
> expiration='2.35.0'

Why do we need to deprecate it in version `2.35.0`?

Reply:
I believe we had picked 2.35 just to give users a few months to redirect to the new command before we actually deprecate it. We introduced the replacement commands (`access-policy [command]` -> `policy [command]`) in this PR with the redirect, and by 2.35.0 we should have all users just using the new `policy [command]` command group.

Is my understanding of `expiration` correct? Will this correctly redirect the users (allowing both commands) from 2.32 -> 2.35, then deprecate the old command and hide it starting in `2.35.0`?

Reply:
Yes, in version `2.35.0`, this command group will be removed directly.